front-end/src/components/views/VideoPlayer.js

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useRef, useState } from 'react';
+import '../../styles/videoPlayerStyles.css';
+import { getDuration } from '../../lib/videoUtils';
+import ReactPlayer from 'react-player';
+
+const VideoPlayer = ({ video, videoTimestamp, setVideoTimestamp }) => {
+
+  const [videoURL, setVideoURL] = useState('');
+
+  useEffect(() => {
+    // Fetch data when the component mounts
+    fetch(`http://${window.location.hostname}:8080/files/${video.key}`)
+      .then((response) => response.blob())
+      .then((blob) => {
+        const url = URL.createObjectURL(blob) 
+        setVideoURL(url);
+
+        return () => {
+          URL.revokeObjectURL(url);
+        };
+      });
+  }, []); // Empty dependency array ensures that the fetch is only performed once
+
+  return (
+    <div className = "background">
```

Comment:
Feels like there are a lot of extra classes / div here that are unneccesary

Reply:
I agree, for one component that is a lot of nested divs